crates/oxc_linter/src/rules/use_isnan.rs

```diff
@@ -0,0 +1,467 @@
+use oxc_ast::{
+    ast::{Argument, ChainElement, Expression},
+    AstKind, GetSpan, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(use-isnan): Requires calls to isNaN() when checking for NaN")]
```

Comment:
nitpick: need some help text I think, eslint has these:

```
        messages: {
            comparisonWithNaN: "Use the isNaN function to compare with NaN.",
            switchNaN: "'switch(NaN)' can never match a case clause. Use Number.isNaN instead of the switch.",
            caseNaN: "'case NaN' can never match. Use Number.isNaN before the switch.",
            indexOfNaN: "Array prototype method '{{ methodName }}' cannot find NaN."
        }
```

Reply:
Done.